lib/src/graph_extension.rs

```diff
@@ -479,6 +480,12 @@ impl GraphExtension for Graph<NodeData, i32> {
         }
     }
 
+    fn set_dag_id(&mut self, dag_id: usize) {
+        for node_i in self.node_indices() {
```

Comment:
Add error handling when the graph has no nodes.

Reply:
Changed to cause panic when the number of nodes is 0.